christfellowshipchurch/web-app
# [Group] Add Group Chat tab to existing Group page

# Related Issue
- [**🏕 Restructure Groups to more closely match mockups for chat**](https://3.basecamp.com/3926363/buckets/17566548/todos/3111116484)

# Changes or Updates
- Adds temporary Chat tab to Group page so it can be shipped independently of wider Group page layout overhaul

# Tradeoffs/Caveats
This isn't a long term layout/solution, just a temporary one so we _can_ ship Group Chat independently of larger Group page layout overhaul work. The tabs are slightly clunky/unrefined when closely examined, but, this is a brief interim state on the way to a much better one shortly.

**🛑 Note** we cannot merge this until the `group-chat` api branch is merged.

#Test Instructions
1. Ensure sandbox mode is **disabled** in your `.env` (`REACT_APP_ENABLE_SANDBOX=`)
1. Ensure you are on the `group-chat` branch on local **🛰️ api**
1. Go to a group page
1. 🔸 Verify you see an **About** and **Chat** tab, and that they work as you'd expect
1. To simulate chat being disabled... open src/groups/GroupContentItemConnected/index.js and remove line 101, where the `chatChannelId` prop is passed down into `<GroupContentItem>`
1. 🔸 Verify the **Chat** tab is gone, and that only **About** is displayed and it is displayed without the blue bottom border

# Screenshots

|State|Screenshot|
|---|---|
|Chat enabled - About Tab|<img src="https://user-images.githubusercontent.com/1812955/96955992-f9154b80-14c4-11eb-9abc-c48c95287571.jpg" width="460" />|
|Chat enabled - Chat Tab|<img src="https://user-images.githubusercontent.com/1812955/96956023-0c281b80-14c5-11eb-8502-088dac0d7dfc.jpg" width="460" />|
|Chat disabled|<img src="https://user-images.githubusercontent.com/1812955/96956030-10543900-14c5-11eb-8c01-cb84b76a8963.jpg" width="460" />|
|Interacting with the Tabs|![2020-10-23 00 05 54](https://user-images.githubusercontent.com/1812955/96955957-e569e500-14c4-11eb-91d7-7abdfa37591d.gif)|


# Tests
